packages/scenario/test/extensibility.test.ts

```diff
@@ -0,0 +1,65 @@
+import { describe, it } from 'vitest';
+
+describe('Extensibility', () => {
+	describe('FormDefTest.java', () => {
+		/**
+		 * **PORTING NOTES**
+		 *
+		 * - It is not inconceivable that we'd support a use case like this. The
+		 *   `xpath` package already supports extensibility of XPath functions (and
+		 *   it utilizes that support to introduce ODK XForms functions). Let's
+		 *   discuss whether this is a use case we expect to support. If so, we
+		 *   should put some serious thought into what the interface for it looks
+		 *   like (including very probably revisiting the `xpath` package's own
+		 *   extensibility APIs, which I'd looked into simplifying in the past. I
+		 *   paused that effort when the functionality which would immediately
+		 *   benefit from it turned out to be out of scope.)
```

Comment:
That's fair. I do think it's at least plausible that if we find a good extensibility API that doesn't incentivize forking, it could be a good way to explore and incubate spec evolutions. And that could be useful both for facilitating third party extensibility (as a typical use case) as well as internal iteration (not unlike the way that real world implementation has been made a formal part of the process for at least some of the web standards bodies).

Reply:
Sounds exciting!